src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -160,29 +196,49 @@ public virtual bool TryFindConnection(string ownerUri, out ConnectionInfo connec
                 };
             }
 
-            // Resolve if it is an existing connection
-            // Disconnect active connection if the URI is already connected
+            // If there is no ConnectionInfo in the map, create a new ConnectionInfo, 
+            // but wait until later when we are connected to add it to the map.
             ConnectionInfo connectionInfo;
-            if (ownerToConnectionMap.TryGetValue(connectionParams.OwnerUri, out connectionInfo) )
+            if (!ownerToConnectionMap.TryGetValue(connectionParams.OwnerUri, out connectionInfo))
```

Comment:
If there is a connection in progress we need to handle this. We should probably return a message stating a connection is in progress and to cancel or wait.